components/remote_settings/src/remotesettings.udl

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+namespace remotesettings {};
+
+[Custom]
+typedef string JsonObject;
+
+dictionary RemoteSettingsConfig {
+    string? server_url;
+    string? bucket_name;
+    string collection_name;
+};
+
+[Error]
+enum RemoteSettingsError {
+    "NetworkFailure",
+    "ClientError",
+    "JSONError",
+    "FileError",
+};
+
+interface RemoteSettings {
+    [Throws=RemoteSettingsError]
+    constructor(RemoteSettingsConfig remote_settings_config);
+
+    [Throws=RemoteSettingsError]
+    sequence<JsonObject> get_records();
```

Comment:
This maybe relates to what Tarik mentioned above - while the records themselves are schemaless, it seems possible there will be metadata you care about - eg, a better return type might be a struct with, say, `last_modified: u64, records: Vec<JsonObject>` or similar. Then you are one step closer to being able to trivially add `get_records_since(u64)` and also have a place to add extra metadata (eg, etag or something else from the server communicated via headers)

Reply:
Latest push has some more significant changes to the data model - much happier with where it is now so thanks for the suggestions. I _think_ I handled the fact that arbitrary JSON can be included with some of these fields like `last_modified` and and `id`, but let me know if you think there are further improvements that can be made